src/components/pages/packages/flyouts/newPackage/newPackage.js

```diff
@@ -1,13 +1,161 @@
+import React from 'react';
+
+import { packageTypeOptions } from 'services/models';
+import { svgs, LinkedComponent, Validator } from 'utilities';
+import {
+  AjaxError,
+  Btn,
+  BtnToolbar,
+  Flyout,
+  FlyoutHeader,
+  FlyoutTitle,
+  FlyoutCloseBtn,
+  FlyoutContent,
+  Indicator,
+  FormControl,
+  FormGroup,
+  FormLabel,
+  SummaryBody,
+  SectionDesc,
+  SummaryCount,
+  SummarySection,
+  Svg
+} from 'components/shared';
 
 import './newPackage.css';
 
+const fileInputAccept = ".json,application/json";
+
 export class NewPackage extends LinkedComponent {
-  // TODO
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      type: undefined,
+      packageFile: undefined,
+      changesApplied: undefined
+    };
+  }
+
+  apply = (event) => {
+    event.preventDefault();
+    const { createPackage } = this.props;
+    const { type, packageFile } = this.state;
+    if (this.formIsValid()) {
+      createPackage({ type: type, packageFile: packageFile });
+      this.setState({ changesApplied: true });
+    }
+  }
+
+  onFileSelected = (e) => {
+    let file = e.target.files[0];
+    this.setState({ packageFile: file });
+  }
+
+  formIsValid = () => {
+    return [
+      this.packageTypeLink,
+    ].every(link => !link.error);
+  }
 
   render() {
-    return ([]);
+    const { t, onClose, isPending, error } = this.props;
+    const { packageFile, changesApplied } = this.state;
+
+    const summaryCount = 1;
+    const typeOptions = packageTypeOptions.map(value => ({
+      label: t(`packages.typeOptions.${value.toLowerCase()}`),
+      value
+    }));
+
+    const completedSuccessfully = changesApplied && !error && !isPending;
+    // Validators
+    const requiredValidator = (new Validator()).check(Validator.notEmpty, t('packages.flyouts.new.validation.required'));
+
+    // Links
+    this.packageTypeLink = this.linkTo('type').map(({ value }) => value).withValidator(requiredValidator);
+
+    return (
+      <Flyout>
+        <FlyoutHeader>
+          <FlyoutTitle>{t('packages.flyouts.new.title')}</FlyoutTitle>
+          <FlyoutCloseBtn onClick={onClose} />
+        </FlyoutHeader>
+        <FlyoutContent className="new-package-content">
+          <form className="new-package-form" onSubmit={this.apply}>
+            <div className="new-package-header">{t('packages.flyouts.new.header')}</div>
+            <div className="new-package-descr">{t('packages.flyouts.new.description')}</div>
+
+            <FormGroup>
+              <FormLabel isRequired="true">{t('packages.flyouts.new.type')}</FormLabel>
+              <FormControl
+                type="select"
+                className="long"
+                link={this.packageTypeLink}
+                options={typeOptions}
+                placeholder={t('packages.flyouts.new.placeHolder')}
+                clearable={false}
+                searchable={false} />
+            </FormGroup>
+
+            <div className="new-package-upload-container">
```

Comment:
Since all these styles are inside new-package-content, you can consider dropping the "new-package" prefix from them.

Reply:
@spryor is it fine to remove  "new-package" prefix?